shopfloor_checkout_package_measurement/services/checkout.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+
+from odoo.addons.base_rest.components.service import to_int
+from odoo.addons.component.core import Component
+from odoo.addons.shopfloor.utils import to_float
+
+
+class Checkout(Component):
+    _inherit = "shopfloor.checkout"
+
+    def _response_for_package_measurement(self, picking, selected_lines, package):
+        packages_data = self.data.package(
+            package,
+            with_packaging=True,
+        )
+        res = self._response(
+            next_state="package_measurement",
+            data={
+                "picking": self.data.picking(picking),
+                "package": packages_data,
+                "package_requirement": self.data.package_requirement(
+                    package.packaging_id
+                ),
+            },
+        )
+        return res
+
+    def set_package_measurement(
+        self,
+        picking_id,
+        package_id,
+        height=None,
+        length=None,
+        width=None,
+        shipping_weight=None,
+    ):
+        """Set measurements on a package."""
+        picking = self.env["stock.picking"].browse(picking_id)
+        message = self._check_picking_status(picking)
+        if message:
+            return self._response_for_select_document(message=message)
+        package = self.env["stock.quant.package"].browse(package_id).exists()
+        if not package:
+            return self._response_for_summary(
+                picking, message=self.msg_store.record_not_found()
+            )
+        if height:
+            package.height = height
+        if length:
+            package.pack_length = length
+        if width:
+            package.width = width
+        if shipping_weight:
+            package.shipping_weight = shipping_weight
+        return self._response_for_summary(
+            picking,
+            message=self.msg_store.package_measurement_changed(),
+        )
+
+    def _put_lines_in_allowed_package(self, picking, selected_lines, package):
+        res = super()._put_lines_in_allowed_package(picking, selected_lines, package)
+        if not picking._check_required_package_measurement(package=package):
```

Comment:
this line has to be adapted to changes in https://github.com/OCA/delivery-carrier/pull/522

Reply:
Yes. But actually the latest changes done in the related PR actually break this one in more ways...
Now that we have the required measurements checked by a constrain this code does not work anymore, because we first save the stock.quant.package (the constrain stops us here) and then check if some of the measurements are required,
One way to get around this would be 
```
    @api.constrains("length", "width", "height", "weight")
    def _check_required_dimension(self, ignore_picking_state=False):
        for package in self:
            domain = [('result_package_id', '=', package.id)]
            pickings = self.env['stock.move.line'].search(domain).mapped('picking_id')  
            if pickings.state != "done" and not ignore_picking_state:
                continue
            required_dimension = []
            if package.length_required and not package.pack_length:
                required_dimension.append(_("length"))
...
```
It can be improve with a read_group but still not that nice...

Reply:
Here is the PR https://github.com/OCA/delivery-carrier/pull/546 to fix that issue...